source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -331,22 +336,55 @@ ResponsePtr RawHttpClientImpl::toResponse(Http::ResponseMessagePtr message) {
     return std::make_unique<Response>(errorResponse());
   }
 
+  // Extract headers-to-remove from the storage header coming from the
+  // authorization server.
+  const auto& storage_header_name = Http::Headers::get().EnvoyAuthHeadersToRemove;
+  // If we are going to construct an Ok response we need to save the
+  // headers_to_remove in a variable first.
+  std::vector<Http::LowerCaseString> headers_to_remove;
+  if (status_code == enumToInt(Http::Code::OK)) {
+    // iterate() over the headers rather than just get() in case the storage
+    // header occurs more than once.
+    absl::string_view storage_header_name_view = absl::string_view(storage_header_name.get());
+    message->headers().iterate([&](const Http::HeaderEntry& entry) {
+      if (entry.key() == storage_header_name_view) {
+        absl::string_view storage_header_value = entry.value().getStringView();
+        std::vector<absl::string_view> header_names = StringUtil::splitToken(
+            storage_header_value, ",", /*keep_empty_string=*/false, /*trim_whitespace=*/true);
+        headers_to_remove.reserve(headers_to_remove.size() + header_names.size());
```

Comment:
nit. Sorry, can we just do `headers_to_remove.reserve(header_names.size());` here?

Reply:
Yes, but the header could be present more than once, so we will hit the `HeaderEntry` more than once:
1. The first time we'll do `reserve(0 + header_names.size())`. 
2. The second time we'll do `reserve(<previous header_name.size()> + header_name.size())`